src/shared/logs-show.c

```diff
@@ -317,7 +317,8 @@ static bool print_multiline(
         return ellipsized;
 }
 
-static int output_timestamp_monotonic(FILE *f, sd_journal *j, const char *monotonic) {
+static int output_timestamp_monotonic(FILE *f, sd_journal *j, OutputMode mode, const char *monotonic) {
+        char buf[CONST_MAX(FORMAT_TIMESTAMP_MAX, 64U)];
```

Comment:
I do not think it is necessary to store string temporary. Why not to call fprintf multiple times?

Reply:
This way we can use `return (int) strlen(buf);`. From my point of view this is a lot easier and less error prone than the `return 1 + 5 + 1 + 6 + 1` which has been used before.

Reply:
And `output_timestamp_realtime()` below is doing the same thing.

Reply:
Changed